src/main/java/org/openstreetmap/atlas/checks/flag/serializer/CheckFlagDeserializer.java

```diff
@@ -51,15 +96,306 @@ public CheckFlag deserialize(final JsonElement json, final Type typeOfT,
         final JsonObject full = json.getAsJsonObject();
         final JsonObject properties = full.get(PROPERTIES).getAsJsonObject();
         final String checkName = properties.get(GENERATOR).getAsString();
-        final String instruction = properties.get(INSTRUCTIONS).getAsString();
-        final String flagIdentifier = properties.get(IDENTIFIERS) == null
```

Comment:
Why is the null check removed?

Reply:
This is removed because it is no longer using the unique identifiers for the new flags identifier. The new flag now will have the same identifier as the old, and the unique identifiers can again be calculated from the flagged objects. Essentially, this was a bit of a hack because everything was not being deserialized previously.